### PR TITLE
Remove matlab ref

### PR DESCRIPTION
**Pull request by sheecegardezi**
rename test data ref from Matlab to legacy
remove test specifically for Matlab funtions
remove dead code
update comments 
remove check for ROI_PAC input data

### CODE REVIEW

**Comment by brenmous** [comment]
The Travis build/unit tests are failing because the ROI_PAC functionality has been removed but the test is running against a ROI_PAC config file, so it's trying to load and treat the ROI_PAC files as GAMMA.

Can you please revert your changes in removing the ROI_PAC support? The last conversation I had with Matt about it was that we have the functionality there so we may as well keep it in case people still want to process ROI_PAC formatted data. We also didn't have any work planned in regards to removing ROI_PAC.

File: pyrate/scripts/run_prepifg.py [hunk 1]
```diff
@@ -75,12 +75,12 @@ def main(params=None):
         if params[cf.APS_ELEVATION_MAP]:
             base_ifg_paths.append(params[cf.APS_ELEVATION_MAP])
 
-    shared.mkdir_p(params[cf.OUT_DIR]) # create output dir
+    shared.mkdir_p(params[cf.OUT_DIR])  # create output dir
     
     process_base_ifgs_paths = \
         np.array_split(base_ifg_paths, mpiops.size)[mpiops.rank]
-    gtiff_paths = [shared.output_tiff_filename(f, \
```
Comment (brenmous) [inline pyrate/scripts/run_prepifg.py]: can you please stop making style changes where it's not needed?

File: pyrate/scripts/run_prepifg.py [hunk 2]
```diff
@@ -112,22 +112,14 @@ def do_prepifg(gtiff_paths, params):
             [_prepifg_multiprocessing(p, xlooks, ylooks, exts, thresh, crop,
                                       params) for p in gtiff_paths]
     else:
-        log.error("Full-res geotiffs do not exist. Ensure you have"
```
Comment (brenmous) [inline pyrate/scripts/run_prepifg.py]: and again here

File: tests/test_mpi.py [hunk 3]
```diff
@@ -180,8 +180,8 @@ def test_vcm_matlab_vs_mpi(mpisync, tempdir, get_config):
 
     maxvar, vcmt = run_pyrate._maxvar_vcm_calc(dest_paths, params_dict,
                                               preread_ifgs)
-    np.testing.assert_array_almost_equal(maxvar, matlab_maxvar, decimal=4)
-    np.testing.assert_array_almost_equal(matlab_vcm, vcmt, decimal=3)
+    np.testing.assert_array_almost_equal(maxvar, legacy_maxvar, decimal=4)
+    np.testing.assert_array_almost_equal(lagecy_vcm, vcmt, decimal=3)
```
Comment (brenmous) [inline tests/test_mpi.py]: typo

File: tests/test_mpi.py [hunk 4]
```diff
@@ -145,12 +145,12 @@ def get_crop(request):
...
 
-def test_vcm_matlab_vs_mpi(mpisync, tempdir, get_config):
+def test_vcm_legacy_vs_mpi(mpisync, tempdir, get_config):
     from tests.common import SML_TEST_DIR, TEST_CONF_ROIPAC
 
     params_dict = get_config(TEST_CONF_ROIPAC)
-    MATLAB_VCM_DIR = os.path.join(SML_TEST_DIR, 'matlab_vcm')
-    matlab_vcm = np.genfromtxt(os.path.join(MATLAB_VCM_DIR, 'matlab_vcmt.csv'), delimiter=',')
+    LEGACY_VCM_DIR = os.path.join(SML_TEST_DIR, 'vcm')
+    lagecy_vcm = np.genfromtxt(os.path.join(LEGACY_VCM_DIR, 'vcmt.csv'), delimiter=',')
```
Comment (brenmous) [inline tests/test_mpi.py]: typo

**Comment by brenmous** [comment]
We don't need to commit the tiff files in test_data/.../gamma_obs/ - these are generated by converttogeotiff as part of the workflow.

File: pyrate/orbital.py [hunk 5]
```diff
@@ -45,10 +45,10 @@
 
 
 # Design notes:
-# The orbital correction code is based on Matlab Pirate, but includes several
-# enhancements. Pirate creates sparse arrays for the linear inversion, which
-# contain many empty cells. This is unnecessary for the independent method, and
-# temporarily wastes potentially a lot of memory.
+# The orbital correction code includes several enhancements. Pirate creates
```
Comment (brenmous) [inline pyrate/orbital.py]: Still have the mention to "Pirate"

File: pyrate/prepifg.py [hunk 6]
```diff
@@ -342,7 +342,7 @@ def _resample_ifg(ifg, cmd, x_looks, y_looks, thresh, md=None):
     """
 
     # Create tmp ifg and extract data array for manual resampling as gdalwarp
-    # lacks the averaging method implemented in Matlab Pirate
+    # lacks the averaging method
```
Comment (brenmous) [inline pyrate/prepifg.py]: might as well pull this line out completely

File: pyrate/timeseries.py [hunk 7]
```diff
@@ -317,7 +315,7 @@ def _solve_ts_lap(nvelpar, velflag, ifgv, mat_b, smorder, smfactor, sel, vcmt):
     tsvel = np.empty(nvelpar, dtype=float32) * np.nan
     tsvel[~np.isclose(velflag, 0.0, atol=1e-8)] = x[:nvelleft]
 
-    # TODO: implement uncertainty estimates (tserror) like in Matlab Pirate code
+    # TODO: implement uncertainty estimates (tserror)
```
Comment (brenmous) [inline pyrate/timeseries.py]: without the reference to matlab this becomes really ambiguous - are you sure we should be nuking every mention of the word "matlab"? Some of these comments being deleted could be useful

File: tests/test_covariance.py [hunk 8]
```diff
@@ -216,16 +214,16 @@ def tearDownClass(cls):
         params = cf.get_config_params(TEST_CONF_ROIPAC)
```
Comment (brenmous) [inline tests/test_covariance.py]: This is why the tests are failing

**Comment by brenmous** [comment]
Along with the above stuff I don't think renaming all the mentions of "Matlab" to "Legacy" achieves anything - that code is still there, people will now just be confused about what legacy refers to

**Comment by sheecegardezi** [comment]
Review failed, changes requested: 

- ROI_PAC functionality need to be added back
- “tiff files in test_data/.../gamma_obs/” need to be added to git ignr
- replace mentions of Pirate with PyRate
- Pull comments from Matlab implementation

File: pyrate/timeseries.py (hunk 7 above)
Comment (sheecegardezi) [reply pyrate/timeseries.py]: I thought the same thing initailly! Just knowing that a function is a translation of a funtion from Pirate dose not tell me what the funtion is ment to do. I made a note in ticket to pull comments from Matlab implementation if available.

**Comment by tfuhrmann** [comment]
I've had a look through the list of changes. I understand that most of the references to Matlab are related to tests, particularly to test whether Matlab PiRate and Python PyRate deliver the same outputs. So, I'm wondering if we should get rid of these tests completely at some stage, since PyRate will be further developed, but PiRate won't. We should have a chat with Matt on this when he's back.

As discussed earlier, I would like to replace the current test data set, which uses Envisat data from 2006/2007, with more recently acquired Sentinel-1 data. This also fits to Alex McVitties's email received on 20/8/2019 asking for a free SBAS test data set. Please add this as a new ticket for the next Sprint.

The only reference for something to be implemented from PiRate into PyRate I found was in pyrate/timeseries.py: 
# TODO: implement uncertainty estimates (tserror)
This can also be added as new ticket for the next (or a later) Sprint. The corresponding PiRate m-file is ts.m.

**Pull request merged**